packages/gatsby-source-shopify/tests/create-operations.spec.ts

```diff
@@ -0,0 +1,41 @@
+import * as createOperationsModule from "../src/create-operations"
```

Comment:
Our convention for tests is:

- `__tests__` folder
- Use snapshots as a last resort

So you'll want to rename/move this file and see if you can turn down the snapshot tests to more programmatic tests

Reply:
You can have a reference here: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-core-utils/src

So:

1. `__mocks__` folder
2. `__tests__` folder
3. `__tests__/fixtures` folder
4. The test files don't have a `.spec` in their name

> I think that snapshots provide valuable insights into the structural changes of data for a PR and should not be removed, but rather supplemented with more rigorous tests.

Yeah, I/We are not saying that all the snapshots should be removed but snapshots with > 3000 lines are not valuable. I think that e.g. https://github.com/gatsbyjs/gatsby/blob/b3c54f2d967223c97d2f39a47f9c0e6e4fd1c0a7/packages/gatsby-source-shopify/__tests__/__snapshots__/create-schema-customization.spec.ts.snap is a test that just copies the SDL that you already wrote to a snapshot file. This is testing the implementation of Gatsby (if it correctly outputs the SDL in combination). I couldn't tell the difference between the different snapshots there. So instead, explicitly assert a selected group of things from that SDL. I'd want to easily read e.g. two tests:

"Ahh, first test asserts that the strings contains X, while the second test asserts that the same line contains Y instead"

That's not something I can see in the snapshots right now.

If this is still all to abstract we should maybe to a sync call on this then :)

Reply:
We don't use `.spec` in the rest of the codebase so please change those filenames.